### PR TITLE
Fix Test Flake [hopefully]

### PR DESCRIPTION
**Pull request by jshaughn**
Fixes https://github.com/kiali/kiali/issues/5358

- ensure kebab is visible before click
- only test span select span one time
  - reduces scrolling which may contribute to the problem
  - reduce unnecessary redundancy/speed up test
- remove some unused imports and variables

### CODE REVIEW

**Comment by jshaughn** [comment]
@nrfox the main test-related changes are just the addition of a `be.visible` and a small optimization of the tests for two tabs for a selected trace.   But there were also a ton of console errors about duplicate keys and I didn't know if maybe that could contribute to a problem, so the other changes are all related to those, adding/changing keys to many places.  I think also some unintended reformatting, just to make it more messy for you.  In short, I think if the tests pass for you locally we can give it a try.  I've seen 3 or 4 straight successes in the CI.

**Comment by nrfox** [comment]
@jshaughn the tests were failing for me locally and it seems to be because there are two different step definitions that are named the same thing:
https://github.com/kiali/kiali/blob/8f36dbbad2332c358370ccb894c311714367d028/frontend/cypress/integration/common/service_details.ts#L84
https://github.com/kiali/kiali/blob/d823380f784ce9649a4ff90d3d06a1b2d18f96aa/frontend/cypress/integration/common/app_details.ts#L75

Probably it chooses one at random if they're both named the same thing. This has been an issue before. Unfortunately there's not a good way of namespacing these or more importantly know when two names overlap.

If both of these step definitions are the same can you consolidate them into one?

**Comment by nrfox** [comment]
Having feature coupled step definitions seems to be an [anti-pattern](https://cucumber.io/docs/guides/anti-patterns/?lang=java#feature-coupled-step-definitions) that we've unfortunately been following.

File: frontend/cypress/integration/common/service_details.ts [hunk 1]
```diff
@@ -13,85 +13,86 @@ Given('a service in the cluster with a healthy amount of traffic', function () {
...
-    cy.get('#pf-tab-section-1-trace-details').within(() => {
-        cy.get('.pf-c-dropdown__toggle.pf-m-plain').first().should('be.visible').click();
-        cy.get('ul.pf-c-dropdown__menu').should('be.visible').contains('Inbound Metrics');
-    });
+    cy.get('table').should('be.visible')
+      .find('tbody tr')                // ignore thead rows
+      .should('have.length.above', 1)  // retries above cy.find() until we have a non head-row
+      .eq(1)                           // take 1st  row
+      .find('td').eq(4)                // take 5th cell (kebab)
+      .click()
```
Comment (nrfox) [inline frontend/cypress/integration/common/service_details.ts:96]: There actually are some helper functions for interacting with tables in [table.ts](https://github.com/kiali/kiali/blob/master/frontend/cypress/integration/common/table.ts#L12-L47) but if this works too or is specific to this test then that's fine.

File: frontend/cypress/integration/common/service_details.ts [hunk 2]
```diff
@@ -13,85 +13,86 @@ Given('a service in the cluster with a healthy amount of traffic', function () {
     this.targetService = SERVICE;
 });
 
-Then('user sees a list with content {string}', (tab: string) => {
+Then('sd::user sees a list with content {string}', (tab: string) => {
```
Comment (nrfox) [inline frontend/cypress/integration/common/service_details.ts:16]: We talked some offline about this. Newer versions of cypress cucumber detect and throw an error for multiple definitions of the same step definition but until we upgrade namespacing will work for keeping them separate.

**Review by nrfox — APPROVED** [review]
LGTM

File: frontend/cypress/integration/common/service_details.ts (hunk 1 above)
Comment (jshaughn) [reply frontend/cypress/integration/common/service_details.ts:96]: Ah, good to know. Thanks.

File: frontend/cypress/integration/common/service_details.ts (hunk 2 above)
Comment (jshaughn) [reply frontend/cypress/integration/common/service_details.ts:16]: This poor-man's namespacing caught several name-clash issues.  The upgrade is still good, although taking this sort of explicit approach makes it very clear that the steps in the associated .ts file are expected to be used.

Comment (jshaughn) [reply frontend/cypress/integration/common/service_details.ts:16]: I'll merge assuming the relevant CI passes.

**Pull request merged**